ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/ExceptionRequestModal/DeferralForm.tsx

```diff
@@ -117,26 +172,53 @@ function DeferralForm({ cves, scopeContext, onCancel }: DeferralFormProps) {
                                             <Radio
                                                 id="custom-date"
                                                 name="custom-date"
-                                                isChecked={false}
-                                                onChange={() => {}}
+                                                isChecked={values.expiry?.type === 'CUSTOM_DATE'}
+                                                onChange={() =>
+                                                    setExpiry({
+                                                        type: 'CUSTOM_DATE',
+                                                        date: addDays(new Date(), 1).toISOString(),
+                                                    })
+                                                }
                                                 label="Until a specific date"
                                             />
                                         )}
-                                        {config.expiryOptions.customDate && false && (
-                                            <div>
-                                                <DatePicker name="custom-date-picker" />
-                                            </div>
-                                        )}
+
+                                        {config.expiryOptions.customDate &&
+                                            values.expiry?.type === 'CUSTOM_DATE' && (
+                                                <div>
+                                                    <DatePicker
+                                                        name="custom-date-picker"
+                                                        value={prettifyDate(values.expiry?.date)}
```

Comment:
For my learning, does TypeScript require optional chaining, given `values.expiry?.type === 'CUSTOM_DATE'` in conditional rendering?